extensions/subscriptions/js/src/forum/addUserPreferences.tsx

```diff
@@ -0,0 +1,50 @@
+import app from 'flarum/forum/app';
+import { extend } from 'flarum/common/extend';
+
+import SettingsPage from 'flarum/forum/components/SettingsPage';
+import FieldSet from 'flarum/common/components/FieldSet';
+import Select from 'flarum/common/components/Select';
+
+import type Mithril from 'mithril';
+
+const keys = ['all_new', 'first_new'];
```

Comment:
I think a boolean field would be easier, `Notify about every new post instead of only the last in a discussion`.

Reply:
I would agree that this makes more sense for this implementation.

I opted to use a select drop down in order to potentially allow for future extensibility by extensions to add their own notification options if they wished. Having said that, though, the way I wrote this doesn't actually allow for that anyway! 😅

Reply:
Yea, the hardest thing about core is always considering extensibility, but considering the slight complexity in this feature, keeping it simple will be beneficial to us 👍🏼.